### PR TITLE
REF: use BlockManager.to_native_types in formatting code

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Small follow-up on https://github.com/pandas-dev/pandas/pull/36150

### CODE REVIEW

File: pandas/core/internals/managers.py [hunk 1]
```diff
@@ -631,6 +631,13 @@ def replace_list(
         bm._consolidate_inplace()
         return bm
 
+    def to_native_types(self, **kwargs) -> "BlockManager":
```
Comment (jbrockmendel) [inline pandas/core/internals/managers.py:634]: more informative name?

Comment (jorisvandenbossche) [reply pandas/core/internals/managers.py:634]: It's the name that the blocks also use..

Comment (jbrockmendel) [reply pandas/core/internals/managers.py:634]: It's not an informative name there either

Comment (jorisvandenbossche) [reply pandas/core/internals/managers.py:634]: Then I am tempted to say "Go for it" .. ;)

I am well aware and fully agree that the whole "native types" terminology is confusing, but it is used in quite some places (not only the Block method, but also the Index helper methods). So fully on board with changing it, but that's a bigger change than just trying to remove the internals leakage in the csv code what I am doing here.

**Comment by jbrockmendel** [comment]
LGTM

File: pandas/core/internals/managers.py (hunk 1 above)
Comment (jreback) [reply pandas/core/internals/managers.py:634]: yeah this can be updated / changed later

**Review by jreback — APPROVED** [review]

**Pull request merged**